pkg/kube/client.go

```diff
@@ -180,18 +183,17 @@ func (c *Client) Update(original, target ResourceList, force bool) (*Result, err
 
 	switch {
 	case err != nil:
-		return nil, err
+		return res, err
 	case len(updateErrors) != 0:
-		return nil, errors.Errorf(strings.Join(updateErrors, " && "))
+		return res, errors.Errorf(strings.Join(updateErrors, " && "))
 	}
 
 	for _, info := range original.Difference(target) {
 		c.Log("Deleting %q in %s...", info.Name, info.Namespace)
+		res.Deleted = append(res.Deleted, info)
```

Comment:
reading the comment in line 193, how come we are appending here before the resource was deleted?

Reply:
This was to match behavior with `Created` and `Updated`. Basically the `Result` should contain everything it _tried_ to create, update, or delete